src/api/services/memory_service.py

```diff
@@ -0,0 +1,378 @@
+import uuid
+import logging
+import hashlib
+import re
+from typing import Optional, Dict, Any
+
+import openai
+from azure.search.documents import SearchClient
+from azure.core.credentials import AzureKeyCredential
+from azure.search.documents.indexes import SearchIndexClient
+from azure.search.documents.indexes.models import (
+    HnswParameters,
+    HnswAlgorithmConfiguration,
+    SemanticPrioritizedFields,
+    SearchableField,
+    SearchField,
+    SearchFieldDataType,
+    SearchIndex,
+    SemanticSearch,
+    SemanticConfiguration,
+    SemanticField,
+    SimpleField,
+    VectorSearch,
+    VectorSearchAlgorithmKind,
+    VectorSearchAlgorithmMetric,
+    ExhaustiveKnnAlgorithmConfiguration,
+    ExhaustiveKnnParameters,
+    VectorSearchProfile,
+)
+
+from common.config.config import Config
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Default memory index name
+MEMORY_INDEX_NAME = "memory-vector"
+
+class MemoryService:
+    """Service for managing memory storage in Azure AI Search."""
+    
+    def __init__(self):
+        config = Config()
+        self.azure_openai_endpoint = config.azure_openai_endpoint
```

Comment:
Are all of these existing resources, or did any new resources need to be provisioned?

Reply:
No, the resources are still the same but just need to change the container so that my local changes reflect :D